### DISCUSSION TITLE
datadir directory does not exist however it does and has permissions

### DISCUSSION THREAD

**Discussion by Crash0v3r1de** [oetiker/SmokePing#315] [Q&A]
So I'm no linux guru here but I do generally understand what I'm doing.

I'm just trying to move the datadir from it's normal default location to librenms /opt/librenms/rrd/smokeping which believe it or not has permissions. However when you start the deamon and try to load the web UI it think's the dir is not writable or not readable although it has complete access (yes it has read and write permissions all the way through to it's final destination dir) but for some reason it think's it's not actually a usable directory but still writes to it.

Not really sure what to do and this is not a LibreNMS issue since the folder is just that - a folder with permissions for smokeping to use.

Here is the deamon status log:
Jan 20 12:33:23 nms systemd[1]: Starting LSB: Start or stop the smokeping latency logging system daemon...
Jan 20 12:33:23 nms smokeping[12156]: Checking smokeping configuration file syntax...
Jan 20 12:33:24 nms smokeping[12156]: ERROR: /etc/smokeping/config.d/General, line 17: Directory 'datadir = /opt/librenms/rrd/smokeping' does not exist
Jan 20 12:33:24 nms systemd[1]: Started LSB: Start or stop the smokeping latency logging system daemon.

I've been trying to figure out why this doesn't realize it in fact has permission and it a valid directory for hours and I've just kind of given up so hoping this is the proper channel for help on this since it is an issue.

**Comment by oetiker** [comment]
It could be that the user, SmokePing is running under does not have permissions to access the datadir in question.

**Reply by Crash0v3r1de** [reply]
> It could be that the user, SmokePing is running under does not have permissions to access the datadir in question.

Yeah I added smokeping user into the required groups for permissions. It most certainly can read and write to the folder and traverse to that folder.

I even got lazy and 777'd opt recursively with the same outcome. (yes that was changed back after that test failed)

**Reply by Crash0v3r1de** [reply]
![image](https://user-images.githubusercontent.com/47544026/150581042-00c61549-f2ab-48a4-9583-404c5810acae.png)

**Reply by oetiker** [reply]
In that case, I don't know what is hitting you ...

**Reply by Crash0v3r1de** [reply]
> In that case, I don't know what is hitting you ...

lol yeah doesn't make any sense to me. Have anything specific I can try to dig into to figure out why this is happening?

**Reply by oetiker** [reply]
well, my question would be if your assumption is correct that smokeping is running in the same security context you are getting when you `su` ... you could try running smokeping by hand, to see if it also runs into trouble or if this is tied to systemd starting it ...

**Reply by Crash0v3r1de** [reply]
> well, my question would be if your assumption is correct that smokeping is running in the same security context you are getting when you `su` ... you could try running smokeping by hand, to see if it also runs into trouble or if this is tied to systemd starting it ...

ran it as root and have the same outcome. Not really sure what to do from here.

**Reply by oetiker** [reply]
maybe there are some special rules in your setup preventing the access for that process ...

**Comment by cnighswonger** [comment]
Try seeing what this says:

```
getfacl /opt/librenms/rrd/smokeping
```
